### ISSUE TITLE
Multiple Set-Cookie headers will be folded when appended in _middleware

### ISSUE THREAD

**Issue by tavvy** [vercel/next.js#34467]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### **Present in Vercel Prod**

It works _fine_ locally, when using `vercel dev` its only once deployed to the genuine edge environment that the issue presents

```
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 20.6.0: Wed Nov 10 22:23:07 PST 2021; root:xnu-7195.141.14~1/RELEASE_X86_64
    Binaries:
      Node: 14.18.3
      npm: 6.14.15
      Yarn: 1.22.15
      pnpm: N/A
    Relevant packages:
      next: 12.0.10
      react: 17.0.2
      react-dom: 17.0.2
```

### Describe the Bug

Despite consulting docs and trying various approaches I cannot set multiple outing going Set-Cookie headers from middleware. Multiple appended `Set-Cookie` headers will be folded into a single header string, delimited by a `,`.

- https://github.com/vercel/next.js/pull/30560
- https://github.com/vercel/next.js/pull/30560/files

I think cloudflare still folds the headers, despite the vercel implementation. See demo below (especially the testing pane)
https://cloudflareworkers.com/?_gl=1*eoxc09*_ga*ODUyNTAyNTk3LjE2MzczMzgwNDc.*_gid*OTcyMTI5ODYxLjE2NDUwMDI0MDc.#a1316a97e1bfe87e7ac74b95032c55d1:https://vercel.com/

### Expected Behavior

I should see multiple `Set-Cookie` response headers.

### To Reproduce

You can reproduce it yourself easily enough. (using NextResponse.cookie just wraps the headers.append call, so using it directly for brevity)

```typescript
import { NextRequest, NextResponse } from "next/server";

export default async function middleware(_request: NextRequest) {
    const resp = NextResponse.next();
    resp.headers.append("Set-Cookie", "cookie1=val1");
    resp.headers.append("Set-Cookie", "cookie2=val2");
    return resp;
}
```

Once deployed to vercel you will see in the response that the headers are concatenated.

![Screenshot 2022-02-17 at 11 32 48](https://user-images.githubusercontent.com/660635/154474009-7e3f0c73-51c0-40e2-975e-c6be9e40488c.png)

**Comment by xhoantran** [comment]
Duplicate of [34254](https://github.com/vercel/next.js/issues/34254)

**Comment by javivelasco** [comment]
A fix was already shipped for this and should hit stable in the coming days so I'm closing this!
